tests/IntegrationTests/Notifications/Repositories/YoutubeRepositoryTests.cs

```diff
@@ -0,0 +1,606 @@
+
+            await repository.AddSubscriptionAsync(channelId, DateTime.Now, defaultGuildId, channelTitle);
+
+            var removeResult = await repository.RemoveSubscriptionAsync(channelId);
+
+            var getResult = await repository.GetSubscriptionByIdOrDefaultAsync(channelId);
+
+            // assert
+			
+            Assert.That(removeResult.IsSuccess, Is.True);
+			Assert.That(getResult, Is.Null);
+		}
+		
+		[Test]
+		public async Task YoutubeRepository_RemoveSubscription_ReturnsMissingSubscriptionError_WhenSubscriptionDoesNotExist() 
+        {
+            // arrange
+
+            var repository = new YoutubeRepository();
+            var channelId = "1";
+
+            // act
+
+            var removeResult = await repository.RemoveSubscriptionAsync(channelId);
+
+            // assert
+            
+            Assert.That(removeResult.IsSuccess, Is.False);
+            Assert.That(removeResult.Error, Is.InstanceOf<MissingSubscriptionError>());
+        }
+		
+		[Test]
+		public async Task YoutubeRepository_TryGetSubscriptionsAsync_ReturnsSubscriptionsCorrectly()
+		{
+            // arrange
+
+            var repository = new YoutubeRepository();
+            var defaultGuildId = "1";
+            var channelId = "1";
+            var channelTitle = "Battle of the Smegheads (HD)";
+
+			// act
+
+            await repository.AddSubscriptionAsync(channelId, DateTime.Now, defaultGuildId, channelTitle);
+
+			var (success, result) = await repository.TryGetSubscriptionsAsync(defaultGuildId);
+
+            // assert
+
+            Assert.That(success, Is.True);
+            Assert.That(result.Count, Is.EqualTo(1));
+			Assert.That(result.First().GuildIds, Has.One.Items.EqualTo(defaultGuildId));
+		}
+		
+		[Test]
+		public async Task YoutubeRepository_TryGetSubscriptionsAsync_ReturnsEmptyList_WhenNoSubscriptionsExist()
+        {
+            // arrange
+
+            var repository = new YoutubeRepository();
+            var channelId = "1";
+
+            // act
+
+            var (success, result) = await repository.TryGetSubscriptionsAsync(channelId);
+
+            // assert
+
+            Assert.That(success, Is.False);
+            Assert.That(result, Is.Empty);
+        }
+		
+		[Test]
+		public async Task YoutubeRepository_UpdateSeenVideos_UpdatesSeenVideosCorrectly()
+		{
+            // arrange
+
+            var repository = new YoutubeRepository();
+            var guildId = "1";
+            var channelId = "1";
+            var channelTitle = "Battle of the Smegheads (HD)";
+
+            var videoId = "1";
+
+			// act
+
+            await repository.AddSubscriptionAsync(channelId, DateTime.Now, guildId, channelTitle);
+
+			await repository.UpdateSeenVideosAsync(channelId, new []{ videoId });
+
+            var getResult = await repository.GetSubscriptionByIdOrDefaultAsync(channelId);
+
+            // assert
+			
+			Assert.That(getResult.AlreadySeenIds, Has.Exactly(1).Items.EqualTo(videoId));
```

Comment:
Missing nullability check, fixes "Dereference of a possibly null reference." warning.
```suggestion
			Assert.That(getResult, Is.Not.Null);
			Assert.That(getResult!.AlreadySeenIds, Has.Exactly(1).Items.EqualTo(videoId));
```